alicloud/resource_alicloud_router_interface.go

```diff
@@ -0,0 +1,297 @@
+			},
+			"description": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateRouterInterfaceDescription,
+			},
+			"health_check_source_ip": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"health_check_target_ip": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceAlicloudRouterInterfaceCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AliyunClient).ecsconn
+	args, err := buildAlicloudRouterInterfaceCreateArgs(d, meta)
+	if err != nil {
+		return err
+	}
+
+	response, err := conn.CreateRouterInterface(args)
+	if err != nil {
+		return fmt.Errorf("CreateRouterInterface got an error: %#v", err)
+	}
+
+	d.SetId(response.RouterInterfaceId)
+	return resourceAlicloudRouterInterfaceUpdate(d, meta)
+}
+
+func resourceAlicloudRouterInterfaceUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AliyunClient).ecsconn
+
+	d.Partial(true)
+	args := &ecs.ModifyRouterInterfaceAttributeArgs{
+		RegionId:          getRegion(d, meta),
+		RouterInterfaceId: d.Id(),
+	}
+	attributeUpdate := false
+
+	if d.HasChange("name") {
+		d.SetPartial("name")
+		args.Name = d.Get("name").(string)
+		attributeUpdate = true
+	}
+
+	if d.HasChange("description") {
+		d.SetPartial("description")
+		args.Description = d.Get("description").(string)
+		attributeUpdate = true
+	}
+
+	if d.HasChange("opposite_router_id") {
+		d.SetPartial("opposite_router_id")
+		args.OppositeRouterId = d.Get("opposite_router_id").(string)
+		attributeUpdate = true
+	}
+
+	if d.HasChange("opposite_interface_id") {
+		d.SetPartial("opposite_interface_id")
+		args.OppositeInterfaceId = d.Get("opposite_interface_id").(string)
+		attributeUpdate = true
+	}
+
+	if d.HasChange("opposite_interface_owner_id") {
+		d.SetPartial("opposite_interface_owner_id")
+		args.OppositeInterfaceOwnerId = d.Get("opposite_interface_owner_id").(string)
+		attributeUpdate = true
+	}
+
+	if d.HasChange("health_check_source_ip") {
+		d.SetPartial("health_check_source_ip")
+		args.HealthCheckSourceIp = d.Get("health_check_source_ip").(string)
+		args.HealthCheckTargetIp = d.Get("health_check_target_ip").(string)
+		attributeUpdate = true
+	}
+
+	if d.HasChange("health_check_target_ip") {
+		d.SetPartial("health_check_target_ip")
+		args.HealthCheckTargetIp = d.Get("health_check_target_ip").(string)
+		args.HealthCheckSourceIp = d.Get("health_check_source_ip").(string)
+		attributeUpdate = true
+	}
+
+	if attributeUpdate {
+		if args.HealthCheckSourceIp != "" && args.HealthCheckTargetIp != "" || args.HealthCheckSourceIp == "" && args.HealthCheckTargetIp == "" {
+			if _, err := conn.ModifyRouterInterfaceAttribute(args); err != nil {
+				return fmt.Errorf("ModifyRouterInterfaceAttribute got an error: %#v", err)
+			}
+		} else {
+			return fmt.Errorf("The 'health_check_source_ip' and 'health_check_target_ip' must set or not set at one time.")
```

Comment:
1. Parameter validate should be check before creating or updating.
2. The 'health_check_source_ip' and 'health_check_target_ip' should be specified or not at one time.